ckanext/emailasusername/tests/test_plugin.py

```diff
@@ -123,3 +124,26 @@ def get_validator_names(validator_list):
         for key, value in default_schema.items():
             assert key in schema
             assert (value in schema[key] or value == schema[key])
+
+    def test_user_registration_frontend_form(self, app):
+        response = app.get(url_for('user.register'))
+        assert 'email2' in response.body
+
+
+@pytest.mark.usefixtures(u'clean_db')
+@pytest.mark.ckan_config(u'ckan.plugins', u'emailasusername')
+@pytest.mark.usefixtures(u'with_plugins')
+@pytest.mark.usefixtures(u'with_request_context')
+@pytest.mark.ckan_config(u'ckanext.emailasusername.require_user_email_input_confirmation', False)
+class TestEmailsWithoutRequiringUserEmailInputConfirmation(object):
+
+    def test_emailasusername_new_user_schema(self):
+        schema = ckan.logic.schema.user_new_form_schema()
+        assert 'email2' not in schema
+        email_validators = get_validator_names(schema['email1'])
+        assert 'user_emails_match' not in email_validators
+        assert 'user_both_emails_entered' not in email_validators
+
+    def test_user_registration_frontend_form(self, app):
+        response = app.get(url_for('user.register'))
+        assert 'email2' not in response.body
```

Comment:
it would make more sense to keep the tests in a signel class. You can user fixtures at test class or method level. So sth like:
```
def test_user_registration_form_defaults_to_email_confirmation(self, app):
        response = app.get(url_for('user.register'))
        assert 'email2' in response.body

def test_default_config_email_input_confirmation_value(self, app):
        assert not config_require_user_email_input_confirmation()

@pytest.mark.ckan_config(u'ckanext.emailasusername.require_user_email_input_confirmation', False)
def test_user_registration_email_confirmation_validatiors(self):
        schema = ckan.logic.schema.user_new_form_schema()
        assert 'email2' not in schema
        email_validators = get_validator_names(schema['email1'])
        assert 'user_emails_match' not in email_validators
        assert 'user_both_emails_entered' not in email_validators

@pytest.mark.usefixtures(u'with_request_context')
@pytest.mark.ckan_config(u'ckanext.emailasusername.require_user_email_input_confirmation', False)
    def test_user_registration_form_disables_email_confirmation(self, app):
        response = app.get(url_for('user.register'))
        assert 'email2' not in response.body
```